docs/user/sbt.rst

```diff
@@ -113,6 +115,32 @@ Garbage collectors
    for short-running command-line applications or applications where garbage
    collections pauses are not acceptable.
 
+Link-Time Optimization (LTO)
+----------------------------
+
+Scala Native relies on link-time optimization to maximize runtime performance
+of release builds. There are three possible modes that are currently supported:
+
+1. **none.** (default)
+
+   Does not inline across Scala/C boundary. Scala to Scala calls
+   are still optimized by emitting one fat LLVM IR module for
+   the whole application.
+
+2. **full.** (Clang 3.8 or older)
+
+   Inlines across Scala/C boundary by merging all of the LLVM IR
+   modules into a single module for the whole application. Unlike
+   **none** this module also includes the runtime code thus allows
+   for additional optimization opportunities.
+
+3. **thin.** (recommended on Clang 3.9 or newer)
+
+   Inlines across Scala/C boundary. Unlike **none** and **full**
```

Comment:
Maybe link to https://clang.llvm.org/docs/ThinLTO.html? For example

```markdown
Inlines across Scala/C boundary by using Clang's
[ThinLTO support](https://clang.llvm.org/docs/ThinLTO.html#clang-llvm)
which supports whole-program analysis and
cross-module optimization. Unlike **none** and **full**
it's able to optimize the application in parallel.
It also offers the best runtime performance according
to our benchmarks.
```

Reply:
Good idea.